metricsaggregator/src/main/java/io/dockstore/metricsaggregator/helper/RunExecutionAthenaAggregator.java

```diff
@@ -0,0 +1,211 @@
+    private static final Field<String> COST_FIELD = field("cost", String.class);
+
+    // Fields for the SELECT clause
+    private final Set<SelectField<?>> selectFields = new HashSet<>();
+    // Fields for the GROUP BY clause
+    private final Set<Field<?>> groupFields = new HashSet<>();
+
+    protected RunExecutionAthenaAggregator(MetricsAggregatorAthenaClient metricsAggregatorAthenaClient, String tableName) {
+        super(metricsAggregatorAthenaClient, tableName);
+        // All queries will be grouped by platform at a minimum
+        selectFields.add(coalesce(PLATFORM_FIELD, inline(Partner.ALL.name())).as(PLATFORM_FIELD.getName())); // Coalesce null platform values to "ALL"
+        groupFields.add(PLATFORM_FIELD);
+    }
+
+    /**
+     * Create a metric from a single query result row.
+     * @param queryResultRow
+     * @return
+     */
+    abstract Optional<M> createMetricFromQueryResultRow(QueryResultRow queryResultRow);
+
+    /**
+     * Get the column name containing the metric.
+     * Athena columns are lowercase, but if a non-lowercase column is provided (ex: camelCase), Athena automatically lowercases it.
+     * @return
+     */
+    abstract String getMetricColumnName();
+
+    public Set<SelectField<?>> getSelectFields() {
+        return this.selectFields;
+    }
+
+    public void addSelectFields(Set<SelectField<?>> newSelectFields) {
+        this.selectFields.addAll(newSelectFields);
+    }
+
+    public Set<Field<?>> getGroupFields() {
+        return this.groupFields;
+    }
+
+    public void addGroupFields(Set<Field<?>> newGroupFields) {
+        this.groupFields.addAll(newGroupFields);
+    }
+
+    /**
+     * Create the runexecutions query string using the SELECT and GROUP BY fields.
+     *
+     * @return
+     */
+    @Override
+    protected String createQuery(AthenaTablePartition partition) {
+        // Sub-query that flattens the runexecutions array for the partition
+        final CommonTableExpression<Record7<String, String, String, Integer, Double, Integer, String>> executionsTable = name("executions")
+                .fields(PLATFORM_FIELD.getName(), DATE_EXECUTED_FIELD.getName(), EXECUTION_STATUS_FIELD.getName(), EXECUTION_TIME_SECONDS_FIELD.getName(),
+                        MEMORY_REQUIREMENTS_GB_FIELD.getName(), CPU_REQUIREMENTS_FIELD.getName(), COST_FIELD.getName())
+                .as(select(PLATFORM_FIELD,
+                        field("unnested." + DATE_EXECUTED_FIELD, String.class),
+                        field("unnested." + EXECUTION_STATUS_FIELD, String.class),
+                        field("unnested." + EXECUTION_TIME_SECONDS_FIELD, Integer.class),
+                        field("unnested." + MEMORY_REQUIREMENTS_GB_FIELD, Double.class),
+                        field("unnested." + CPU_REQUIREMENTS_FIELD, Integer.class),
+                        field("unnested." + COST_FIELD, String.class))
```

Comment:
Thought for the future, wonder if we'll have to be more sophisticated about cost. A string also may make difficult to process via Athena but we may have to deal with different currencies.
Can deal with it when we have (the good problem) to deal with it

Reply:
Specifying this as a `String` doesn't affect the resulting query for cost. This query is just for flattening the RunExecutions array and this line will resolve to `unnested.cost` and will have the form of a `Cost` object (with value and currency).

The cost aggregator currently only takes into account the cost value, but like you mentioned, we may get more sophisticated with our cost aggregation later. When this happens, the change will occur somewhere in this file so it takes into account `cost.currency` for example  https://github.com/dockstore/dockstore-support/blob/bf345fd1aeb613e913e98165fe8c4c969261c05b/metricsaggregator/src/main/java/io/dockstore/metricsaggregator/helper/CostAthenaAggregator.java#L18